File: app/src/main/java/workshop1024/com/xproject/model/message/source/MessageDataSource.java
package workshop1024.com.xproject.model.message.source;

import java.util.List;

import workshop1024.com.xproject.model.message.Message;
import workshop1024.com.xproject.model.message.MessageGroup;

public interface MessageDataSource {
    void getMessages(LoadMessagesCallback loadMessagesCallback);

    void submitMessage(Message message);

    interface LoadMessagesCallback {
        void onMessagesLoaded(List<MessageGroup> messageGroupList);

        void onDataNotAvailable();
    }
}

File: app/src/main/java/workshop1024/com/xproject/model/message/source/MessageDataSource.kt
package workshop1024.com.xproject.model.message.source

import workshop1024.com.xproject.model.message.Message
import workshop1024.com.xproject.model.message.MessageGroup

interface MessageDataSource {
    fun getMessages(loadMessagesCallback: LoadMessagesCallback)

    fun submitMessage(message: Message)

    interface LoadMessagesCallback {
        fun onMessagesLoaded(messageGroupList: List<MessageGroup>)

        fun onDataNotAvailable()
    }
}
